packages/flutter_tools/lib/src/build_runner/build_runner.dart

```diff
@@ -279,3 +293,14 @@ class _BuildRunnerCodegenDaemon implements CodegenDaemon {
     buildDaemonClient.startBuild();
   }
 }
+
+// Sorts the builders by name and produces a hashcode of the resulting iterable.
+String _produceScriptId(YamlMap builders) {
+  if (builders == null || builders.isEmpty) {
+    return '0';
+  }
+  final List<String> orderedBuilders = builders.keys
+    .cast<String>()
+    .toList()..sort();
+  return hashObjects(orderedBuilders).toString();
```

Comment:
Is it possible to reuse https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/base/fingerprint.dart for this? Hash codes are not very reliable w.r.t. collisions.